src/app/fdctl/run/tiles/fd_dedup.c

```diff
@@ -16,8 +16,17 @@
    checks the transaction signature field for duplicates and filters
    them out. */
 
-#define GOSSIP_IN_IDX (0UL) /* Frankendancer and Firedancer */
-#define VOTER_IN_IDX  (1UL) /* Firedancer only */
+#ifdef FD_HAS_NO_AGAVE
```

Comment:
Not the correct way to do this ... why can't you keep the current mechanism?

Reply:
> why can't you keep the current mechanism?

do you mean:
```
#define GOSSIP_IN_IDX (0UL) /* Frankendancer only */
#define VOTER_IN_IDX  (0UL) /* Firedancer only */
```

Without topology awareness we need to make the check for gossip link optional. The current check (in `unprivileged_init`) makes it mandatory for both topologies. I didn't want to break that assumption so I went this way. But if it's not actually a strict requirement I can change that. 

I'm also curious: what's the correct way to weave in topology-aware blocks of code?